modules/stochastic_tools/src/transfers/PODSamplerSolutionTransfer.C

```diff
@@ -0,0 +1,234 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// StochasticTools includes
+#include "PODSamplerSolutionTransfer.h"
+#include "NonlinearSystemBase.h"
+#include "Sampler.h"
+
+registerMooseObject("StochasticToolsApp", PODSamplerSolutionTransfer);
```

Comment:
I renamed this from SamplerSolutionTransfer to PODSamplerSolutionTransfer because even if the general concept is not POD specific, the implementation is. It works only with a PODFullSolveMultiapp and it checks it in the constructor:

```
auto pod_pointer = std::dynamic_pointer_cast<PODFullSolveMultiApp>(_multi_app);
  if (pod_pointer)
    _pod_multi_app = pod_pointer;
  else
    paramError("multi_app", "The Multiapp given is not a PODFullsolveMultiapp!");
```

It also needs a specific function from the PODMultiApp to see if the snapshots have already been obtained or not:
```
if (!_pod_multi_app->snapshotGeneration())
{
...
}
```
This is all necessary for batch-mode. Therefore, I would like to keep POD in the name for now, when the more general class is ready, we can change the name. I think it is very specific to POD right now.